include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -2325,1147 +2325,1161 @@ AffineConstraints<number>::distribute(VectorType &vec) const
     };
 
-  private:
-    /**
-     * The actual data object that contains a scratch data for each thread.
-     */
-    static Threads::ThreadLocalStorage<ScratchData> scratch_data;
-  };
 
+    // function for block matrices: Find out where in the list of local dofs
+    // (sorted according to global ids) the individual blocks start.
+    // Transform the global indices to block-local indices in order to be
+    // able to use functions like vector.block(1)(block_local_id), instead of
+    // vector(global_id). This avoids transforming indices one-by-one later
+    // on.
+    template <typename number, class BlockType>
+    inline void
+    make_block_starts(const BlockType &            block_object,
+                      GlobalRowsFromLocal<number> &global_rows,
+                      std::vector<size_type> &     block_starts)
+    {
+      AssertDimension(block_starts.size(), block_object.n_block_rows() + 1);
 
-  // function for block matrices: Find out where in the list of local dofs
-  // (sorted according to global ids) the individual blocks start.
-  // Transform the global indices to block-local indices in order to be
-  // able to use functions like vector.block(1)(block_local_id), instead of
-  // vector(global_id). This avoids transforming indices one-by-one later
-  // on.
-  template <typename number, class BlockType>
-  inline void
-  make_block_starts(const BlockType &            block_object,
-                    GlobalRowsFromLocal<number> &global_rows,
-                    std::vector<size_type> &     block_starts)
-  {
-    AssertDimension(block_starts.size(), block_object.n_block_rows() + 1);
+      using row_iterator         = std::vector<Distributing>::iterator;
+      row_iterator block_indices = global_rows.total_row_indices.begin();
 
-    using row_iterator         = std::vector<Distributing>::iterator;
-    row_iterator block_indices = global_rows.total_row_indices.begin();
+      const size_type num_blocks    = block_object.n_block_rows();
+      const size_type n_active_rows = global_rows.size();
 
-    const size_type num_blocks    = block_object.n_block_rows();
-    const size_type n_active_rows = global_rows.size();
+      // find end of rows.
+      block_starts[0] = 0;
+      for (size_type i = 1; i < num_blocks; ++i)
+        {
+          row_iterator first_block = Utilities::lower_bound(
+            block_indices,
+            global_rows.total_row_indices.begin() + n_active_rows,
+            Distributing(block_object.get_row_indices().block_start(i)));
+          block_starts[i] = first_block - global_rows.total_row_indices.begin();
+          block_indices   = first_block;
+        }
+      block_starts[num_blocks] = n_active_rows;
+
+      // transform row indices to block-local index space
+      for (size_type i = block_starts[1]; i < n_active_rows; ++i)
+        global_rows.global_row(i) =
+          block_object.get_row_indices()
```

Comment:
Are we not indenting these files with clang-format? The old version looks wrong.

Reply:
Oh, I see - GitHub renders the rest weirdly when we hide whitespace changes. Nevermind!